File: Core/SyncTaskExemption.vb
<Serializable>
Public Class SyncTaskExemption

    Private _entity As ExemptionEntity
    Private _operator As ExemptionOperator
    Private _value As String

    Public Sub New()

    End Sub

    Public Sub New(ByVal entity As ExemptionEntity, _
                   ByVal [operator] As ExemptionOperator, _
                   ByVal value As String)
        Me.Entity = entity
        Me.Operator = [operator]
        Me.Value = value
    End Sub

#Region "PROPERTIES"

    Public Property Entity As ExemptionEntity
        Get
            Return _entity
        End Get
        Set(value As ExemptionEntity)
            _entity = value
        End Set
    End Property

    Public Property [Operator] As ExemptionOperator
        Get
            Return _operator
        End Get
        Set(value As ExemptionOperator)
            _operator = value
        End Set
    End Property

    Public Property Value As String
        Get
            Return _value
        End Get
        Set(value As String)
            _value = value
        End Set
    End Property

#End Region

#Region "METHODS"

    Public Function Clone() As SyncTaskExemption
        Dim item As New SyncTaskExemption
        With item
            .Entity = Me.Entity
            .Operator = Me.Operator
            .Value = Me.Value
        End With
        Return item
    End Function

#End Region

End Class

File: BP.Unify.Core/SyncTaskExemption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BP.Unify.Core
{
    class SyncTaskExemption
    {
		private ExemptionEntity _entity;
		private ExemptionOperator _operator;
		private string _value;

		public SyncTaskExemption()
		{

		}

		public SyncTaskExemption(ExemptionEntity entity, ExemptionOperator @operator, string value)
		{
			this.Entity = entity;
			this.Operator = @operator;
			this.Value = value;
		}

		#region PROPERTIES

		public ExemptionEntity Entity
		{
			get { return _entity; }
			set { _entity = value; }
		}

		public ExemptionOperator @Operator
		{
			get { return _operator; }
			set { _operator = value; }
		}

		public string Value
		{
			get { return _value; }
			set { _value = value; }
		}

		#endregion

		#region METHODS

		public SyncTaskExemption Clone()
		{
			SyncTaskExemption item = new SyncTaskExemption();
			item.Entity = this.Entity;
			item.Operator = this.Operator;
			item.Value = this.Value;
			return item;
		}

		public override string ToString()
		{
			if (this.Entity != ExemptionEntity.FileSize)
			{
				return Common.FormattedExemptionEntities[this.Entity] + " " + Common.FormattedExemptionOperators[this.Operator] + " " + this.Value;
			}
			else
			{
				return Common.FormattedExemptionEntities[this.Entity] + " " + Common.FormattedExemptionOperators[this.Operator] + " " + Int32.Parse(this.Value).ToString("N0") + " KB";
			}
		}

		#endregion
	}
}
